src/vs/workbench/browser/parts/views/customView.ts

```diff
@@ -796,9 +796,26 @@ class TreeRenderer extends Disposable implements ITreeRenderer<ITreeItem, FuzzyS
 			templateData.resourceLabel.setResource({ name: label, description }, { title, hideIcon: true, extraClasses: ['custom-view-tree-node-item-resourceLabel'], matches: matches ? matches : createMatches(element.filterData) });
 		}
 
-		templateData.icon.style.backgroundImage = iconUrl ? DOM.asCSSUrl(iconUrl) : '';
 		templateData.icon.title = title ? title : '';
 		DOM.toggleClass(templateData.icon, 'custom-view-tree-node-item-icon', !!iconUrl);
+
+		let codicon: string | undefined;
+		if (iconUrl?.scheme === 'icon' && iconUrl?.authority === 'vscode.codicons') {
+			codicon = `codicon-${escape(iconUrl.path.substr(1))}`;
+		}
+
+		DOM.toggleClass(templateData.icon, 'codicon', !!codicon);
+		templateData.icon.classList.forEach((cls, i, list) => {
```

Comment:
@jrieken I wasn't sure if I should have extracted this in to DOM and get it to work with our custom classList model or not. Since this is in the workbench, I figured we don't need to support IE, so the native classList is supported.

Reply:
Yeah, I think that's OK. My hope is that in January, when Edge is public, we drop IE support 🤞An alternative to the classname cleanup might be a dedicated span that's only used for codicon on which you can set/unset a single classname.